### ISSUE TITLE
End test upon failure

### ISSUE THREAD

**Issue by justingosan** [substack/tape#367]
Is there a built-in or reliable way to end running the succeeding tests upon failing one of them?

Thanks,
Justin

**Comment by ljharb** [comment]
We have an onFinish callback, but not an onFailure callback.

That seems like a reasonable thing to add.

**Comment by olalonde** [comment]
Yes, something like `mocha --bail` would be nice. 

```
-b, --bail                      bail after first test failure
```

**Comment by olalonde** [comment]
Just found https://github.com/juliangruber/tap-bail

**Comment by kevinpeno** [comment]
@olalonde Sorry if this is a silly question, but how'd you set this up? I tried using is as `tape test/**/*.js | tap-bail` (got a pipe error) and `tape --tap test/**/*.js | tap-bail` and got no output.

Other notes, I'd like to use `blue-tape` and `faucet` with this, but I disabled them for this test.

**Comment by olalonde** [comment]
@kevinpeno you should probably ask on that repo as I just started using it :P my script is `babel-tape-runner test/custom.js | tap-bail | faucet`

**Comment by lastmjs** [comment]
Can this be closed now?

**Comment by justingosan** [comment]
Closed! Thanks everyone!

**Comment by ljharb** [comment]
Fixed in #408.
